### PR TITLE
[FLINK-13221][Table SQL / Planner] Blink planner should set ScheduleMode to LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST for batch jobs

### PR DESCRIPTION
**Pull request by liyafan82**
## What is the purpose of the change

Make batch jobs use LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST as the default scheduling mode.

## Brief change log

  - Replace LAZY_FROM_SOURCES with LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST for batch jobs.
  

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java [hunk 1]
```diff
@@ -366,7 +366,7 @@ private JobGraph createBatchJobGraph() {
 		v2.connectNewDataSetAsInput(v1, DistributionPattern.ALL_TO_ALL, ResultPartitionType.BLOCKING);
 
 		final JobGraph jobGraph = new JobGraph(v1, v2);
-		jobGraph.setScheduleMode(ScheduleMode.LAZY_FROM_SOURCES);
+		jobGraph.setScheduleMode(ScheduleMode.LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST);
```
Comment (KurtYoung) [inline flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java]: unnecessary changes?

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobgraph/ScheduleModeTest.java [hunk 2]
```diff
@@ -31,6 +31,7 @@
 	@Test
 	public void testAllowLazyDeployment() throws Exception {
 		assertTrue(ScheduleMode.LAZY_FROM_SOURCES.allowLazyDeployment());
+		assertTrue(ScheduleMode.LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST.allowLazyDeployment());
```
Comment (KurtYoung) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobgraph/ScheduleModeTest.java]: unnecessary changes?

File: flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java [hunk 3]
```diff
@@ -117,7 +117,7 @@ public void testHandleStreamingJobsWhenNotEnoughSlot() throws Exception {
 	@Test
 	public void testHandleBatchJobsWhenNotEnoughSlot() throws Exception {
 		try {
-			setupAndRunHandleJobsWhenNotEnoughSlots(ScheduleMode.LAZY_FROM_SOURCES);
+			setupAndRunHandleJobsWhenNotEnoughSlots(ScheduleMode.LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST);
```
Comment (KurtYoung) [inline flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java]: unnecessary changes?

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java [hunk 4]
```diff
@@ -593,12 +593,19 @@ public void testDefaultScheduleMode() {
...
 			.setScheduleMode(ScheduleMode.LAZY_FROM_SOURCES)
 			.generate();
-		JobGraph jobGraph = StreamingJobGraphGenerator.createJobGraph(streamGraph);
-		assertEquals(ScheduleMode.LAZY_FROM_SOURCES, jobGraph.getScheduleMode());
+		JobGraph jobGraph1 = StreamingJobGraphGenerator.createJobGraph(streamGraph1);
+		assertEquals(ScheduleMode.LAZY_FROM_SOURCES, jobGraph1.getScheduleMode());
+
+		StreamGraph streamGraph2 = new StreamGraphGenerator(Collections.emptyList(),
+			env.getConfig(), env.getCheckpointConfig())
+			.setScheduleMode(ScheduleMode.LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST)
```
Comment (KurtYoung) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java]: unnecessary changes?

**Comment by liyafan82** [comment]
> Thanks @liyafan82 for working on this. I think this change should stick to blink planner, we don't have to modify other runtime codes.

@KurtYoung I see. Thanks a lot for your comments.

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java (hunk 1 above)
Comment (liyafan82) [reply flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java]: Reverted. Thanks.

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobgraph/ScheduleModeTest.java (hunk 2 above)
Comment (liyafan82) [reply flink-runtime/src/test/java/org/apache/flink/runtime/jobgraph/ScheduleModeTest.java]: Reverted. Thanks.

File: flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java (hunk 3 above)
Comment (liyafan82) [reply flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java]: Reverted. Thanks.

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java (hunk 4 above)
Comment (liyafan82) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java]: Reverted. Thanks.
